### PR TITLE
NOISSUE - Add Twins to docker-compose and nginx conf

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

### CODE REVIEW

File: docker/nginx/nginx-key.conf [hunk 1]
```diff
@@ -116,6 +116,12 @@ http {
             proxy_pass http://opcua-adapter:${MF_OPCUA_ADAPTER_HTTP_PORT};
         }
 
+        # Proxy pass to mainflux-twins
+        location /twins {
```
Comment (bbokun) [inline docker/nginx/nginx-key.conf]: location ~ ^/(twins|states) {

as the states of the twins also have to be loaded from the twins service

File: docker/nginx/nginx-x509.conf [hunk 2]
```diff
@@ -126,6 +126,12 @@ http {
             proxy_pass http://opcua-adapter:${MF_OPCUA_ADAPTER_HTTP_PORT};
         }
 
+        # Proxy pass to mainflux-twins
+        location /twins {
```
Comment (bbokun) [inline docker/nginx/nginx-x509.conf]: location ~ ^/(twins|states) 

the states of the twins have to read from the twins service

File: src/app/pages/admin/twins/details/twins.details.component.html [hunk 3]
```diff
@@ -55,85 +83,134 @@
...
+          </div>
+          <div class="col-2">
             <nb-checkbox (checkedChange)="togglePersist($event)" [checked]="editAttr.persist_state"></nb-checkbox>
           </div>
         </div>
 
-        <button (click)="addAttribute()" nbButton status="success" size="tiny">
-          Add/Update attribute
+        <button (click)="addAttribute()" nbButton size="small">
+          Add / Update attribute
```
Comment (bbokun) [inline src/app/pages/admin/twins/details/twins.details.component.html]: somehow I prefer Add/Update (without blanks)

File: src/app/pages/things/lora/lora.component.ts [hunk 4]
```diff
@@ -104,7 +104,7 @@ export class LoraComponent implements OnInit {
   offset = 0;
   limit = 20;
 
-  searchFreq = 0;
+  searcTime = 0;
```
Comment (bbokun) [inline src/app/pages/things/lora/lora.component.ts]: use searchTime

File: src/app/pages/things/lora/lora.component.ts [hunk 5]
```diff
@@ -199,9 +199,9 @@ export class LoraComponent implements OnInit {
 
   searchLora(input) {
     const t = new Date().getTime();
-    if ((t - this.searchFreq) > defFreq) {
+    if ((t - this.searcTime) > defSearchBardMs) {
       this.getLoraDevices(input);
-      this.searchFreq = t;
+      this.searcTime = t;
```
Comment (bbokun) [inline src/app/pages/things/lora/lora.component.ts]: searchTime

File: src/app/pages/things/opcua/opcua.component.ts [hunk 6]
```diff
@@ -130,7 +129,7 @@ export class OpcuaComponent implements OnInit {
   offset = 0;
   limit = 20;
 
-  searchFreq = 0;
+  searcTime = 0;
```
Comment (bbokun) [inline src/app/pages/things/opcua/opcua.component.ts]: searchTime please

File: src/app/pages/things/opcua/opcua.component.ts [hunk 7]
```diff
@@ -273,9 +272,9 @@ export class OpcuaComponent implements OnInit {
 
   searchNode(input) {
     const t = new Date().getTime();
-    if ((t - this.searchFreq) > defFreq) {
+    if ((t - this.searcTime) > defSearchBardMs) {
```
Comment (bbokun) [inline src/app/pages/things/opcua/opcua.component.ts]: searchTime

File: src/app/pages/things/opcua/opcua.component.ts [hunk 8]
```diff
@@ -273,9 +272,9 @@ export class OpcuaComponent implements OnInit {
...
       this.getOpcuaNodes(input);
-      this.searchFreq = t;
+      this.searcTime = t;
```
Comment (bbokun) [inline src/app/pages/things/opcua/opcua.component.ts]: searchTime

File: src/app/pages/admin/twins/details/twins.details.component.html (hunk 3 above)
Comment (manuio) [reply src/app/pages/admin/twins/details/twins.details.component.html]: it's not readable in the button...

File: src/app/pages/things/lora/lora.component.ts (hunk 4 above)
Comment (manuio) [reply src/app/pages/things/lora/lora.component.ts]: It's what I'm using

File: .env [hunk 9]
```diff
@@ -193,3 +193,20 @@ MF_POSTGRES_READER_DB_SSL_MODE=disable
...
+MF_TWINS_DB_HOST=twins-db
+MF_TWINS_DB_PORT=27018
+MF_TWINS_SINGLE_USER_EMAIL=""
+MF_TWINS_SINGLE_USER_TOKEN=""
+MF_TWINS_CLIENT_TLS=""
+MF_TWINS_CA_CERTS=""
+MF_TWINS_MQTT_URL=tcp://mqtt-adapter:1883
+MF_TWINS_THING_ID=8041bbca-34d0-435f-86c1-21798555b59a
+MF_TWINS_THING_KEY=65c0244b-b4ad-4286-a9de-74a19de20b45
+MF_TWINS_CHANNEL_ID=b3cfd2cc-ad49-4ed8-82b7-cb80e544acdb
```
Comment (bbokun) [inline .env]: we had a discussion about this to remove this things in an .env-twins file as this is not in the standard. What do You think about that?

File: src/app/pages/admin/twins/details/twins.details.component.html (hunk 3 above)
Comment (bbokun) [reply src/app/pages/admin/twins/details/twins.details.component.html]: Ok, if that is the case, then it is ok for me

File: .env (hunk 9 above)
Comment (manuio) [reply .env]: This is planed to be created in the service itself and removed from here. Even if it's in a separate .env it have to be used by the same docker-compose so i don't really see any benefit.

Comment (bbokun) [reply .env]: Ok

**Pull request merged**